harness/determined/cli/workspace.py

```diff
@@ -232,22 +244,30 @@ def create_workspace(args: argparse.Namespace) -> None:
         defaultComputePool=args.default_compute_pool,
         defaultAuxPool=args.default_aux_pool,
     )
-    if args.namespace:
-        # When args.cluster_name is unspecified, the string "null" assumes its value in the
-        # clusterNamespacePairs dictionary. To avoid that, we substitute the argument's value with
-        # the empty string if it's null.
-        cluster_name = args.cluster_name or ""
-        namespace_meta = bindings.v1WorkspaceNamespaceMeta(namespace=args.namespace)
+
+    cluster_name = args.cluster_name or ""
+    requested_namespace = args.namespace or ""
+    if args.namespace or args.auto_create_namespace:
+        # When args.cluster_name and/or args.namespace is unspecified, the string "null" assumes its
+        # value in the clusterNamespacePairs dictionary. To avoid that, we substitute the argument's
+        # value with the empty string if it's null.
```

Comment:
Thanks for adding the comment! Can you move this above `cluster_name = args.cluster_name or ""`